x/mongo/driver/operation/count.go

```diff
@@ -102,16 +127,63 @@ func (c *Count) Execute(ctx context.Context) error {
 		ServerAPI:         c.serverAPI,
 	}.Execute(ctx, nil)
 
+	// Swallow error if NamespaceNotFound(26) is returned from aggregate on non-existent namespace
+	if err != nil {
+		dErr, ok := err.(driver.Error)
+		if ok && dErr.Code == 26 {
+			err = nil
+		}
+	}
+	return err
 }
 
 func (c *Count) command(dst []byte, desc description.SelectedServer) ([]byte, error) {
-	dst = bsoncore.AppendStringElement(dst, "count", c.collection)
+	// If wire version < 12 (4.9.0), use count command
+	if desc.WireVersion.Max < 12 {
+		dst = bsoncore.AppendStringElement(dst, "count", c.collection)
+		if c.maxTimeMS != nil {
+			dst = bsoncore.AppendInt64Element(dst, "maxTimeMS", *c.maxTimeMS)
+		}
+		if c.query != nil {
+			dst = bsoncore.AppendDocumentElement(dst, "query", c.query)
+		}
+		return dst, nil
+	}
+	// If wire version >= 12 (4.9.0), use aggregate with $collStats
+	header := bsoncore.Value{Type: bsontype.String, Data: bsoncore.AppendString(nil, c.collection)}
+	if c.collection == "" {
+		header = bsoncore.Value{Type: bsontype.Int32, Data: []byte{0x01, 0x00, 0x00, 0x00}}
+	}
+
+	dst = bsoncore.AppendValueElement(dst, "aggregate", header)
+	cursorIdx, cursorDoc := bsoncore.AppendDocumentStart(nil)
```

Comment:
In this `aggregate` command, the `cursor` sub-document will always be empty because it only contains cursor-related options like `batchSize`, which aren't supported by the `EstimatedDocumentCount` helper. Given this, I think we can simplify by removing the temporary sub-document:

```
idx, dst := bsoncore.AppendDocumentElementStart(dst, "cursor")
dst, _ = bsoncore.AppendDocumentEnd(dst, idx)
```